src/HealthGPS/analysis_module.cpp

```diff
@@ -326,28 +326,64 @@ void AnalysisModule::calculate_population_statistics(RuntimeContext &context,
     // Calculate in-place averages
     for (auto i = min_age; i <= max_age; i++) {
```

Comment:
I know this is out of scope for this PR, but this could be better vectorised. Not sure if performance is an issue here though...

Reply:
Could just add TBB parallel_for in all the outer loops right now.